### DISCUSSION TITLE
Bullet 3 is a deterministic physics

### DISCUSSION THREAD

**Discussion by hejiang123** [bulletphysics/bullet3#3821] [General]
Hi, I'm going to use bullet3 physics engine to make a frame synchronization game. But I'm not sure if bullet3 is a deterministic physics engine. I need to get the same results on different platforms

**Comment by erwincoumans** [comment]
Yes, Bullet can be deterministic. Bullet is only deterministic on 1 platform/compiler. The Tiny Differentiable Simulator can be made deterministic, using fixed-point math, but it is very limited in features. See https://github.com/google-research/tiny-differentiable-simulator

**Reply by hejiang123** [reply]
Thank you for your reply. If I want to change the floating-point number of Bullet to a fixed-point number, will it be troublesome to execute and fetch it? Because I currently feel that Bullet is very suitable for my needs. But it's just a matter of fixed-point numbers. I don’t know how much work is needed to modify this piece. Can you give me some appropriate suggestions?

**Reply by erwincoumans** [reply]
zero change, hence tiny-differentiable-simulator.

**Reply by hejiang123** [reply]
Still don't understand how I need to replace btscalar with fix64scalar, or do I just need to integrate without replacing it? Can you tell me a little bit about the details? Thank you very much
